gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/Orchestrator.java

```diff
@@ -133,6 +137,31 @@ public AddSpecResponse onAddSpec(Spec addedSpec) {
     return new AddSpecResponse<>(null);
   }
 
+  /*
+    validates if lease can be acquired on the provided flowSpec,
+    else throw LeaseUnavailableException
+   */
+  private void validateAdhocFlowLeasability(FlowSpec flowSpec) {
+    if (!flowSpec.isScheduled()) {
+      Config flowConfig = flowSpec.getConfig();
+      String flowGroup = flowConfig.getString(ConfigurationKeys.FLOW_GROUP_KEY);
+      String flowName = flowConfig.getString(ConfigurationKeys.FLOW_NAME_KEY);
+
+      DagActionStore.DagAction dagAction = DagActionStore.DagAction.forFlow(flowGroup, flowName,
+          FlowUtils.getOrCreateFlowExecutionId(flowSpec), DagActionStore.DagActionType.LAUNCH);
+      DagActionStore.LeaseParams leaseParams = new DagActionStore.LeaseParams(dagAction, System.currentTimeMillis());
+      _log.info("validation of lease acquirability of adhoc flow with lease params: " + leaseParams);
+      try {
+        if (!dagManagementStateStore.isLeaseAcquirable(leaseParams)) {
+          throw new LeaseUnavailableException("Lease already occupied by another execution of this flow");
+        }
+      } catch (IOException exception) {
+        _log.error(String.format("Failed to query leaseArbiterTable for existing flow details: %s", flowSpec), exception);
```

Comment:
we called `dagManagementStateStore.isLeaseAcquirable(leaseParams)`... who said anything about "leaseArbiterTable"? :)

(anyway, the table's name is dynamically set in config).

instead:
```
_log.error("unable to check whether lease acquirable " + leaseParams, ex);
```

(also on the line below)